File: Source/IoT/src/main/java/com/example/IoT/dto/response/telemetry/TelemetryResponse.java
package com.example.IoT.dto.response.telemetry;

import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Builder
public class TelemetryResponse {
    private String deviceId;
    private String name;
    private String value;
    private String unit;
    private Boolean status;
}

File: Source/App Fire Alarm/app/src/main/java/com/example/firealarm/data/model/telemetry/TelemetryResponse.kt
package com.example.firealarm.data.model.telemetry

import com.example.firealarm.domain.model.Telemetry

data class TelemetryResponse(
    val deviceId: String,
    val name: String,
    val value: String?,
    val unit: String?,
    val status: Boolean?
)

fun TelemetryResponse.toDomain(): Telemetry {
    return Telemetry(
        deviceId = deviceId,
        name = name,
        value = value,
        unit = unit,
        status = status
    )
}

